src/common/walFilter/postgres_common.h

```diff
@@ -0,0 +1,260 @@
+#ifndef PGBACKREST_POSTGRES_COMMON_H
+#define PGBACKREST_POSTGRES_COMMON_H
+
+#include <stdint.h>
+
+// types
+typedef uint32_t TimeLineID;
+typedef uint64_t XLogRecPtr;
+typedef uint8_t RmgrId;
+typedef uint32_t pg_crc32;
+typedef unsigned int Oid;
+typedef uint32_t TransactionId;
+typedef uint32_t BlockNumber;
+
+// constants
+#define XLOG_BLCKSZ 32768
+/* This flag indicates a "long" page header */
+#define XLP_LONG_HEADER             0x0002
+#define MAXIMUM_ALIGNOF 8
+#define BLCKSZ 32768
+#define XLR_MAX_BKP_BLOCKS      4
+#define XLP_FIRST_IS_CONTRECORD     0x0001
+#define XLP_FIRST_IS_OVERWRITE_CONTRECORD 0x0008
+#define XLR_INFO_MASK           0x0F
+#define XLOG_HEAP_OPMASK        0x70
+#define GPDB6_XLOG_PAGE_MAGIC 0xD07E
+
+// macros
+#define MAXALIGN(LEN)           TYPEALIGN(MAXIMUM_ALIGNOF, (LEN))
+#define TYPEALIGN(ALIGNVAL,LEN)  \
+    (((uintptr_t) (LEN) + ((ALIGNVAL) - 1)) & ~((uintptr_t) ((ALIGNVAL) - 1)))
+
+#define SizeOfXLogShortPHD  MAXALIGN(sizeof(XLogPageHeaderData))
+#define SizeOfXLogLongPHD   MAXALIGN(sizeof(XLogLongPageHeaderData))
+#define XLogPageHeaderSize(hdr)     \
+    (((hdr)->xlp_info & XLP_LONG_HEADER) ? SizeOfXLogLongPHD : SizeOfXLogShortPHD)
+#define SizeOfXLogRecord    MAXALIGN(sizeof(XLogRecord))
+#define XLogRecGetData(record)  ((char*) (record) + SizeOfXLogRecord)
+#define XLR_BKP_BLOCK(iblk)     (0x08 >> (iblk))        /* iblk in 0..3 */
+#define Min(x, y)       ((x) < (y) ? (x) : (y))
+
+// enums
+// need to know size
+typedef enum ForkNumber
+{
+    InvalidForkNumber = -1,
+    MAIN_FORKNUM = 0,
+    FSM_FORKNUM,
+    VISIBILITYMAP_FORKNUM,
+
+    /*
+     * Init forks are used to create an initial state that can be used to
+     * quickly revert an object back to its empty state. This is useful for
+     * reverting unlogged tables and indexes back to their initial state during
+     * recovery.
+     */
+    INIT_FORKNUM
+
+    /*
+     * NOTE: if you add a new fork, change MAX_FORKNUM and possibly
+     * FORKNAMECHARS below, and update the forkNames array in
+     * src/common/relpath.c
+     */
+} ForkNumber;
```

Comment:
using headers from the installed version of gpdb is not the best idea, since one pgbackrest build should be able to support different versions of gpdb.